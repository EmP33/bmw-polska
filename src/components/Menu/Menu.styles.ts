import styled from "styled-components";

export const Container = styled.div<{ showMenu: boolean }>`
  width: 100%;
  height: 100vh;
  background: var(--color-black);
  color: var(--color-white);
  position: absolute;
  left: 0;
  top: 0;
  z-index: 1000;
  display: grid;
  grid-template-rows: max-content 1fr max-content;
  align-items: center;
  justify-items: center;
  transform: ${(props) =>
    props.showMenu ? "translateX(0)" : "translateX(-100%)"};
  transition: all 0.2s linear;

  .menu-action {
    width: 100%;
    display: grid;
    grid-template-columns: 1fr max-content;
    column-gap: 3rem;
    row-gap: 1rem;
    padding: 1rem 2rem 0 0;

    & > div {
      border-bottom: 1px solid #343434;
    }

    svg {
      width: 40px;
      height: 40px;
      grid-column: 2/-1;
      margin-bottom: 1rem;
    }

    button {
      grid-column: 2/-1;
      font-size: 3rem;
      background: none;
      border: none;
      color: #858585;
      display: flex;
      align-items: center;
      justify-content: center;
      width: 40px;
      height: 40px;
      cursor: pointer;

      &:hover {
        filter: brightness(80%);
      }
    }
  }

  ul {
    list-style: none;
    text-align: center;
    width: 70%;
    margin: 0 auto;

    & > li {
      font-family: var(--font-heading);
      font-weight: 600;
      padding: 2rem 0 2rem 0;
      cursor: pointer;

      &:hover {
        color: var(--color-gray-6);
      }

      &:not(:last-child) {
        border-bottom: 1px solid #343434;
      }
    }
  }

  & > button {
    margin-bottom: 5rem;
  }
`;
