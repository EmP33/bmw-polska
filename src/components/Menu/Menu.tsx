import React from "react";
// Styles
import { Container } from "./Menu.styles";
import { PrimaryButton } from "../../Components.styles";
import { BmwLogo } from "../../assets";

interface Props {
  toggleMenu: () => void;
  showMenu: boolean;
}

const Menu: React.FC<Props> = ({ toggleMenu, showMenu }) => {
  return (
    <Container showMenu={showMenu}>
      <div className="menu-action">
        <div></div>
        <BmwLogo />
        <button type="button" onClick={toggleMenu}>
          &times;
        </button>
      </div>
      <ul>
        <li>
          <a onClick={toggleMenu} href="#info">
            NOWE BMW iX
          </a>
        </li>
        <li>
          <a onClick={toggleMenu} href="#info2">
            INNOWACYJNA AERODYNAMIKA
          </a>
        </li>
        <li>
          <a onClick={toggleMenu} href="#gallery">
            GALERIA
          </a>
        </li>
      </ul>
      <PrimaryButton onClick={toggleMenu}>
        <a href="#contact">UMÓW JAZDĘ PRÓBNĄ</a>
      </PrimaryButton>
    </Container>
  );
};

export default Menu;
