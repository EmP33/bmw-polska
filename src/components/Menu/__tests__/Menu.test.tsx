import Menu from "../Menu";
import ReactShallowRenderer from "react-test-renderer/shallow";

describe("<Menu />", () => {
  it("should render properly", () => {
    const renderer = new ReactShallowRenderer();
    const toggleMenu = jest.fn();
    renderer.render(<Menu showMenu={true} toggleMenu={toggleMenu} />);
    expect(renderer.getRenderOutput()).toMatchSnapshot();
  });
});
