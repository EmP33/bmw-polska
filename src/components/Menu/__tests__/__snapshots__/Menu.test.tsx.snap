// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Menu /> should render properly 1`] = `
<styled.div
  showMenu={true}
>
  <div
    className="menu-action"
  >
    <div />
    <BmwLogo />
    <button
      onClick={[MockFunction]}
      type="button"
    >
      ×
    </button>
  </div>
  <ul>
    <li>
      <a
        href="#info"
        onClick={[MockFunction]}
      >
        NOWE BMW iX
      </a>
    </li>
    <li>
      <a
        href="#info2"
        onClick={[MockFunction]}
      >
        INNOWACYJNA AERODYNAMIKA
      </a>
    </li>
    <li>
      <a
        href="#gallery"
        onClick={[MockFunction]}
      >
        GALERIA
      </a>
    </li>
  </ul>
  <styled.button
    onClick={[MockFunction]}
  >
    <a
      href="#contact"
    >
      UMÓW JAZDĘ PRÓBNĄ
    </a>
  </styled.button>
</styled.div>
`;
